writer.go

```diff
@@ -76,3 +76,33 @@ func open(paths []string) ([]zapcore.WriteSyncer, func(), error) {
 	}
 	return writers, close, errs.AsError()
 }
+
+// OpenWriters combines the passed io.Writer objects that are compatible with
+// zapcore.WriteSyncer into a locked WriteSyncer. It returns any errors
+// encountered.
+func OpenWriters(writers ...io.Writer) (zapcore.WriteSyncer, error) {
+	w := make([]zapcore.WriteSyncer, 0, len(writers))
+
+	for _, writer := range writers {
+		writeSyncer, ok := writer.(zapcore.WriteSyncer)
+		if ok {
+			w = append(w, writeSyncer)
+		}
+	}
+
+	return OpenWriteSyncers(w...)
+}
+
+// OpenWriteSyncers combines the passed set of WriteSyncer objects into a locked
+// WriteSyncer. It returns any error encountered
+func OpenWriteSyncers(writers ...zapcore.WriteSyncer) (zapcore.WriteSyncer, error) {
+	if len(writers) == 0 {
+		return zapcore.AddSync(ioutil.Discard), nil
+	}
+
+	if len(writers) == 1 {
+		return zapcore.Lock(writers[0]), nil
+	}
```

Comment:
Let's actually make this improvement to `zapcore.NewMultiWriteSyncer` - if it's passed just one argument, it should return that `WriteSyncer` unchanged. This wrapper can then rely on that behavior.